budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/contract/overview/table/TotalContractDetailsModel.java

```diff
@@ -0,0 +1,87 @@
+package org.wickedsource.budgeteer.web.pages.contract.overview.table;
+
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.IObjectClassAwareModel;
+import org.apache.wicket.model.LoadableDetachableModel;
+import org.apache.wicket.model.Model;
+import org.joda.money.Money;
+import org.wickedsource.budgeteer.MoneyUtil;
+import org.wickedsource.budgeteer.service.contract.ContractTotalData;
+import org.wickedsource.budgeteer.service.contract.ContractBaseData;
+import org.wickedsource.budgeteer.service.contract.DynamicAttributeField;
+import org.wickedsource.budgeteer.web.pages.budgets.overview.table.TotalBudgetDetailsModel;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.wicketstuff.lazymodel.LazyModel.from;
+import static org.wicketstuff.lazymodel.LazyModel.model;
+
+public class TotalContractDetailsModel extends LoadableDetachableModel<ContractTotalData> implements IObjectClassAwareModel<ContractTotalData> {
+    private IModel<List<ContractTotalData>> wrappedModel;
+
+    public TotalContractDetailsModel(IModel<List<ContractBaseData>> sourceModel){
+        List<ContractTotalData> totalDataList = new ArrayList<>();
+
+        // Cast given sourceModel to a model of ContractTotalData.
+        for(ContractBaseData baseData: sourceModel.getObject())
+        {
+            totalDataList.add(new ContractTotalData(baseData));
+        }
+
+        wrappedModel = model(from(totalDataList));
+    }
+
+    /**
+     * Creates a list of emtpy strings.
+     * The length of the list is equal to the number of the contract attributes of a contract.
+     * @return a list of empty strings
+     */
+    public List<String> getContractAttributes()
```

Comment:
Why do you need a list of empty strings?

Reply:
This was also needed for the dynamically added contract attributes. Because I don't now all the features of wicket yet, my first approach was use a ListView, so I needed a List in the right size.
I changed this part, so now it uses a RepeatingView, which seems to fit better here.